### ISSUE TITLE
Fails to work with 2022.6.2: SQL error: `UNION types text and integer cannot be matched`

### ISSUE THREAD

**Issue by OmenWild** [ryanwinter/hass-rainforest-emu-2#17]
I tried upgrading from 2022.5.5 to 2022.6.2 last night, but this addon caused a SQL error and was unable to read any data from the device. Every 30 seconds or so I would get a `ERROR (MainThread) [custom_components.rainforest_emu_2.emu2] issued command while not connected` error. I have attached the log snipped from HA for the SQL error that may be the cause. The error is `UNION types text and integer cannot be matched`, and  checking my HA install, I can see that `"4482c9ec491fe5b4f5c4ff4b2c997d3c"` is my "Rainforest EMU-2" device.

Other than this issue, the addon has been awesome! Thank you for making it.

[home-assistant-1.log](https://github.com/ryanwinter/hass-rainforest-emu-2/files/8845905/home-assistant-1.log)

**Comment by ryanwinter** [comment]
Hi @OmenWild,

Yes this is an erroneous error which occurs because the Home Assistant is trying to send to the device because the device is ready. There is probably some additional code I can put do to tell Hom Assistant that the device isnt connected yet.

Is this causing any issues with the data coming through?

**Comment by OmenWild** [comment]
Hi Ryan, sorry that was incomplete reporting on my end. I believe all of the data points were missing (`unavailable`) in HA, but I forgot to take a screenshot. If I get some spare time I will try upgrading again and take a screenshot.

 Are you running 2022.6.2 successfully?

**Comment by ryanwinter** [comment]
Oh ok, if you are seeing this error continuously (and not just a one off at startup), then this probably means that its not able to connect to the device. I think I need some better connection logging here if thats the case.

**Comment by ryanwinter** [comment]
Oh I'm only running 2022.6.0 and this version is working for me. I'm going to upgrade to see if I'm seeing any issue.

**Comment by ryanwinter** [comment]
Double checking and 2022.6.2 is working for me at this time.

**Comment by OmenWild** [comment]
This is very odd, I upgraded to 2022.6.5, and initially I got a bunch of `ERROR (MainThread) [custom_components.rainforest_emu_2.emu2] issued command while not connected` errors, but no SQL errors like I was getting before. I let HA run for about 5 minutes and got 24 of those errors, and never got any values is HA (all `Unknown`).

Then I restarted HA and it started working correctly, so I am confused. I guess I will close this with #CantReplicate .
